zerver/actions/realm_settings.py

```diff
@@ -621,6 +638,7 @@ def do_delete_all_realm_attachments(realm: Realm, *, batch_size: int = 1000) ->
         obj_class._default_manager.filter(realm=realm).delete()
 
 
+@transaction.atomic(durable=True)
 def do_scrub_realm(realm: Realm, *, acting_user: UserProfile | None) -> None:
```

Comment:
The addition of `@transaction.atomic` was suggested by Tim [over here](https://github.com/zulip/zulip/pull/26109#discussion_r1325231157).